monai/apps/reconstruction/networks/nets/utils.py

```diff
@@ -131,3 +134,43 @@ def complex_normalize(x: Tensor) -> Sequence:  # type: ignore
         )
         x = x.view(b, c, h, w, d)
         return (x - mean) / std, mean, std  # type: ignore
+
+
+def sens_reduce(x: Tensor, sens_maps: Tensor, spatial_dims: int = 2) -> Tensor:
+    """
+    Reduces coil measurements to a corresponding image based on the given sens_maps. Let's say there
+    are C coil measurements inside x, then this function multiplies the conjugate of each coil sensitivity map with the
+    corresponding coil image. The result of this process will be C images. Summing those images together gives the
+    resulting "reduced image."
+
+    Args:
+        x: 2D kspace (B,C,H,W,2) with the last dimension being 2 (for real/imaginary parts) and C denoting the
+            coil dimension. 3D data will have the shape (B,C,H,W,D,2).
+        sens_maps: sensitivity maps of the same shape as input x.
+        spatial_dims: is 2 for 2D data and is 3 for 3D data
+
+    Returns:
+        reduction of x to (B,1,H,W,2) for 2D data or (B,1,H,W,D,2) for 3D data.
+    """
+    x = ifftn_centered_t(x, spatial_dims=spatial_dims)  # inverse fourier transform
+    return complex_mul(x, complex_conj(sens_maps)).sum(dim=1, keepdim=True)  # type: ignore
+
+
+def sens_expand(x: Tensor, sens_maps: Tensor, spatial_dims: int = 2) -> Tensor:
+    """
+    Expands an image to its corresponding coil images based on the given sens_maps. Let's say there
+    are C coils. This function multiples image x with each coil sensitivity map in sens_maps and stacks
+    the resulting C coil images along the channel dimension which is reserved for coils.
+
+    Args:
+        x: 2D image (B,1,H,W,2) with the last dimension being 2 (for real/imaginary parts). 3D data will have
```

Comment:
If x is an image, would it be better to change it to `img`?

Reply:
yeah, we could use that... and also `ksp` for the `sensitivity_map_reduce` function.